src/Arr.php

```diff
@@ -109,11 +109,13 @@ public static function updateIn(array $array, array $path, callable $updater): a
             }
 
             if (!$atDepth) {
+                /* @phpstan-ignore-next-line */
                 return $up($m[$k]); // keep looking
             }
 
             $m[$k] = $updater($m[$k]);
 
+            /* @phpstan-ignore-next-line */
```

Comment:
phpstan was struggling with return type on the anonymous function. trying to docblock it didn't work.

Reply:
@johncorrelli I pulled this down to tinker with it for a minute.

I was able to make it pass by separating the return from the expression:

```php
        $up = function (&$m) use ($updater, &$path, &$up, &$depth, $target) {
            if (empty($path) || !is_array($m)) {
                return;
            }
            $k = array_shift($path);
            $atDepth = ++$depth === $target;

            if (!isset($m[$k])) {
                $m[$k] = []; // path is invalid
            }

            if (!$atDepth) {
                $up($m[$k]); // keep looking

                return;
            }

            $m[$k] = $updater($m[$k]);

            $up($m[$k]);

            return;
        };
```

For brevity's sake, we were doing it on a single line, even though we were never interested in the return value. I think the extra line is preferable to `ignore-next-line` biz. I'll let you push up the change if you agree 🙇🏻

Reply:
I like it... working on it right now. :heart: